newrelic/data_source_newrelic_synthetics_secure_credential.go

```diff
@@ -0,0 +1,55 @@
+package newrelic
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceNewRelicSyntheticsSecureCredential() *schema.Resource {
```

Comment:
For my own education, is there anything here that marks this a data source other than the file name?  This looks pretty similar to the resources below.

Reply:
A data source only has a Read function, and the Schema excludes fields that are not returned (in this case `value` is not returned by the API, so it's not included in the schema).  I agree that there is a bit of duplication here, but not sure how to remove it without adding complexity to reading the code.

Reply:
@zlesnr It's wired up in `newrelic/provider.go` in the collection of data sources, rather than in the collection of resources.